rs/boundary_node/rate_limits/canister/interface.did

```diff
@@ -0,0 +1,111 @@
+  description: opt text;             // Textual description of the rule, none if the rule is currently confidential
+  disclosed_at: opt Timestamp;       // Timestamp when the rule was disclosed, none if the rule is still confidential
+};
+
+type OutputConfig = record {
+  rules: vec OutputRule;
+};
+
+// Response structure for returning the requested configuration and associated metadata
+type OutputConfigResponse = record {
+    version: Version;         // Version of the configuration
+    active_since: Timestamp;  // Time when this configuration was added (became active)
+    config: OutputConfig;     // Contains the list of rules
+};
+
+// Verbose details of an individual rule
+// Optional rule_raw and description fields are for restricted publicly viewing access
+type OutputRuleMetadata = record {
+  id: RuleId;                      // Unique identifier for the rule
+  rule_raw: opt blob;              // Raw rule data (binary format), expected to be a valid json object, none if the rule is currently confidential
+  description: opt text;           // Textual description of the rule, none if the rule is currently confidential
+  disclosed_at: opt Timestamp;     // Timestamp when the rule was disclosed, none if the rule is still confidential
+  added_in_version: Version;       // Version when the rule was added (became active)
+  removed_in_version: opt Version; // Version when the rule was deactivated (removed), none if the rule is still active
+};
+
+type GetRuleByIdResponse = variant {
+  Ok: OutputRuleMetadata;
+  Err: text;
+};
+
+type GetConfigResponse = variant {
+  Ok: OutputConfigResponse;
+  Err: text;
+};
+
+type DiscloseRulesResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type OverwriteConfigResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type DiscloseRulesByRuleIdsResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type DiscloseRulesByIncidentIdResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type GetRulesByIncidentIdResponse = variant {
+  Ok: vec RuleId;
+  Err: text;
+};
+
+// Configuration containing a list of rules that replaces the current configuration
+type InputConfig = record {
+  schema_version: SchemaVersion; // schema version used to serialized the rules
+  rules: vec InputRule;
+};
+
+// Initialization arguments for the service
+type InitArg = record {
+  registry_polling_period_secs: nat64; // IDs of existing API boundary nodes are polled from the registry with this periodicity
+};
+
+service : (InitArg) -> {
+  // Replaces the current rate-limit configuration with a new set of rules
+  overwrite_config: (InputConfig) -> (OverwriteConfigResponse);
+
+  // Make the viewing of the specified rules publicly accessible
+  disclose_rules_by_rule_ids: (vec RuleId) -> (DiscloseRulesByRuleIdsResponse);
+
+  // Make the viewing of the specified rules related to an incident ID publicly accessible
+  disclose_rules_by_incident_id: (IncidentId) -> (DiscloseRulesByIncidentIdResponse);
```

Comment:
Any thoughts on consolidating these two disclose methods? e.g
```
type DiscloseArg = variant {
  Rules: vec RuleId;
  Incidents: vec IncidentId;
};

disclose: (vec DiscloseArg) -> (DiscloseResponse);
```
and similarly for `get_rules`:
```
type ListArg = variant {
  Ids: vec RuleId;
  Incidents: vec IncidentId;
};

list: (vec ListArg) -> (ListResponse);
```

Reply:
I do like the `disclose` one! but for `get_rule**s**` it would be strange for a `RuleId`, as it returns a single rule.